### ISSUE TITLE
1.3.2 Multiplayer Server crash & steps to resolve

### ISSUE THREAD

**Issue by VallenFrostweaver** [adam9899/Issues#192]
Sadly I can't determine the issue but here are the logs from the stream we opened the server to multiple players. Adam was there for it so I won't go into too much detail.
https://hastebin.com/baluzonaju.rb
https://hastebin.com/aranumuqam.rb
https://hastebin.com/kotorozeca.rb
We are looking at it too and will report back with any results we come up with. At a glance we're guessing that it's too many chunks loaded at once.

**Comment by VallenFrostweaver** [comment]
Follow up: Turns out it's a massive amount of entities being loaded at once.
Steps I've taken:
1. Disable Quark frogs (big lag issue from this cute thing)
2. Disable Quark "Animals eat floor food" (less AI issues)
3. Disable Quark "Horse whistle" (was on every entity ticking it)
4. Disable Champions "Plague" potion affix by deleting it from configs (spread was terrible)
5. Disable Multimob "Summoner" from spawning (both Multimob & Primitive Mobs configs)
6. Disable Primitive mobs "Chameleon" from spawning (both Multimob & Primitive Mobs configs)
7. Disable Quark "Items flash before expiring" (it would re-tick things needlessly)
8. Crossed my fingers. (this doesn't help)
I'll report again on the results this may/may not have on the server.

**Comment by VallenFrostweaver** [comment]
UPDATE: The server has been running much better with less crashes. We are still experimenting and searching with the issues happening with more than 2 players online. The Champions mod dev is updating to allow Plague to be removed in configs as it's being ignored and still present. Quark has an issue causing a lot of issues that is going to need a patch. We are testing that patch. MalisisDoors is the current issue we see where it is doing collision checks for doors on practically everything... so, we are disabling MalisisDoors in hopes this will fix a few more issue. Another update to come soon with more info and suggestions.

**Comment by VallenFrostweaver** [comment]
UPDATE: Disabled MalisisDoors & MalisisCore and tried new Quark patch (plus all above of 1-8 list). Results: From 7-8 TPS with 4 players to 16-18 TPS with 4 players. Also just disabled spider families from both Multimob & Primitive Mobs configs as they spawn in groups of entities and can spawn in groups of groups making it very heavy at times.
ISSUES: Disabling MalisisDoors/Core may have had a drastic help on the server but it nullifies 2 quests in the Introduction quests tab.
I hope this has helped. I'll update again if we find more and if Quark adds in that patch.
NOTE: It should be noted that Primitive Mobs is still a heavy mod for servers as it ignores spawn rules. More players = more entities and kills servers over time. Pretty sure you've seen our issues on stream @adam9899 and I hope all this info helps. If you need more or want to chat you can reach me on Discord (Vallen#0002) as I don't always get messages here.

**Comment by adam9899** [comment]
Perfect thanks for the detailed report Vallen. I'll look into limiting Primitive Mobs spawns for definite, it's controlled by the MultiMob config and the numbers are inflated at the moment, needs some proper tweaking for large multiplayer experiences. Malisis doors is a known tps culprit, it's worth disabling the vanilla door configuration if you wanted to keep it enabled.

**Comment by matthew-williams** [comment]
Also changing fast leaf decay to 
```
main {
    # Maximum time in ticks for leaf decay. Must be higher than MinimumDecayTime!
    I:MaximumDecayTime=20

    # Minimum time in ticks for leaf decay. Must be lower than MaximumDecayTime!
    I:MinimumDecayTime=5
}
```
Helps significantly with steve's carts tree farms. Or when veinmining trees

**Comment by jsheldon** [comment]
@VallenFrostweaver Thanks for this.  This was incredibly helpful as I'm running into issues running a server as well.   If you don't mind, could you tell me the spec's on your server?  I'm currently running an 8gig of memory vps from a recommended hosting company.  Was going to try your suggestions, as we hit pretty terrible TPS at 6 players currently.

**Comment by VallenFrostweaver** [comment]
Ours is a 10gig private. We are still having issues though. Crashes and aged rollbacks as the saves need to happen more often but we aren't sure we can find too many more solutions.

**Comment by matthew-williams** [comment]
![Lag Goggles](https://user-images.githubusercontent.com/6619443/74201525-48616780-4c27-11ea-96a2-3a635c7eb095.PNG)
Looks like Primative mobs Grove sprites are also top of the list.

**Comment by matthew-williams** [comment]
Also was seeing quite a few log errors revolving around Entities:

> Fetching addPacket for removed entity

This was the solution: [https://github.com/Daveyx0/PrimitiveMobs/issues/192](url)

**TLDR**; just set the mobs to 0 in weight section of MM and disable them all in PM and MM configs

MM= MultiMob
PM = PrimativeMobs

Current disabled mob list:
- GroveSprites
- Summoners
- Chameleons

**Comment by VallenFrostweaver** [comment]
UPDATE: Our crashes from Primitve mobs and TheSummoner got bad. We have straight up disabled those 2 as well as Multimob. We had players do the same. Most remaining lag was Myrmex spawns so we lowered that too in the Ice & Fire configs. It's been pretty playable since.

**Comment by matthew-williams** [comment]
Can confirm removing those three mods turned our 3 person 24gb RAM server from 11tps to 20tps

Current top Lag Causers:
- Item Frames in villages. Lots spawn and cause massive lag in certain huts.
- Vampires moving weirdly, causing lag when they fight you. More than one is virtually unkillable just cause you lag so bad.

**Comment by VallenFrostweaver** [comment]
UPDATE: We turned off Harvestcraft's crop spread and turned down the Myrmex spawn/laying/etc. and the server has been good since.

**Comment by NLBlacky** [comment]
We did the above mentioned changes (and adjusted Rats configs as well) and all went alot smoother. We did re-enable the Malis door config option since it caused purple door textures when disabled. Thanks for your information regarding performance tweaks.

**Comment by blairwilson** [comment]
Thank you so much - this helps a ton!

Changes we made to Rats.cfg too:
```
Rats will conduct expensive CPU operations like looking for crops or chests, once every this number of ticks(with added standard deviation for servers) [range: 1 ~ 10000, default: 100]
    I:"Rat Upgrade Delay"=200

    # Distance that untamed rats will immediately despawn. For Vanilla mobs, this is 128 blocks. [range: 0.0 ~ 3.4028235E38, default: 96.0]
    S:"Rat Immediate Despawn Distance"=96

    # When a rat is farther than its potential despawn distance, a random roll is taken to see if it despawns. Lower this number will make rats more likely to despawn. [range: 1 ~ 2147483647, default: 400]
    I:"Rat Potential Despawn Chance"=200

    # Distance that untamed rats could possibly despawn. For Vanilla mobs, this is 32 blocks. [range: 0.0 ~ 3.4028235E38, default: 20.0]
    S:"Rat Potential Despawn Distance"=20.0

    # The weight of rats in vanilla's spawn rate [range: 1 ~ 300, default: 80]
    I:"Rat Spawn Weight"=25
```

Also considered disabling Ratlantis entirely as config says it offers alternate methods of getting the mods items but I don't want to go full blitz on the server. Yet. ;D
